scripts/create-agent.ts

```diff
@@ -271,19 +271,19 @@ AWS_DEFAULT_REGION=us-east-1`;
 
       writeFileSync('/tmp/agent.env', envContent);
       const setupCommands = [
-        'sudo apt update',
-        'sudo apt install -y python3.12 python3.12-venv python3-pip',
-        'sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.12 1',
-        'sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.12 1',
-        'curl -sSL https://install.python-poetry.org | python - -y --version 1.8.3',
-        'source ~/.profile'
+        { tag: 'APT', command: 'sudo apt update' },
+        { tag: 'APT', command: 'sudo apt install -y python3.12 python3.12-venv python3-pip' },
+        { tag: 'PYTHON', command: 'sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.12 1' },
+        { tag: 'PYTHON', command: 'sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.12 1' },
+        { tag: 'POETRY', command: 'curl -sSL https://install.python-poetry.org | python - -y --version 1.8.3' },
+        { tag: 'SHELL', command: 'source ~/.profile' }
       ];
       
       const keyPath = `${tmpdir()}/${keyPairName}.pem`;
       
-      for (const command of setupCommands) {
-        console.log(`Executing: ${command}`);
-        await this.executeSSHCommand(keyPath, instanceDetails.publicDns, command);
+      for (const commandObj of setupCommands) {
+        console.log(`Executing [${commandObj.tag}]: ${commandObj.command}`);
+        await this.executeSSHCommand(keyPath, instanceDetails.publicDns, commandObj.command, commandObj.tag);
```

Comment:
just pass in the full `commandObj` and move the console log from this loop as the first line of this method